src/Symfony/Component/Console/Tests/DependencyInjection/AddConsoleCommandPassTest.php

```diff
@@ -0,0 +1,116 @@
+
+use Symfony\Component\Console\DependencyInjection\AddConsoleCommandPass;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+class AddConsoleCommandPassTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider visibilityProvider
+     */
+    public function testProcess($public)
+    {
+        $container = new ContainerBuilder();
+        $container->setResourceTracking(false);
+        $container->addCompilerPass(new AddConsoleCommandPass());
+        $container->setParameter('my-command.class', 'Symfony\Component\Console\Tests\DependencyInjection\MyCommand');
+
+        $definition = new Definition('%my-command.class%');
+        $definition->setPublic($public);
+        $definition->addTag('console.command');
+        $container->setDefinition('my-command', $definition);
+
+        $container->compile();
+
+        $id = $public ? 'my-command' : 'console.command.symfony_component_console_tests_dependencyinjection_mycommand';
+        $this->assertTrue($container->hasParameter('console.command.ids'));
+        $this->assertSame(array($id), $container->getParameter('console.command.ids'));
+    }
+
+    public function visibilityProvider()
+    {
+        return array(
+            array(true),
+            array(false),
+        );
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The service "my-command" tagged "console.command" must not be abstract.
+     */
+    public function testProcessThrowAnExceptionIfTheServiceIsAbstract()
+    {
+        $container = new ContainerBuilder();
+        $container->setResourceTracking(false);
+        $container->addCompilerPass(new AddConsoleCommandPass());
+
+        $definition = new Definition('Symfony\Component\Console\Tests\DependencyInjection\MyCommand');
+        $definition->addTag('console.command');
+        $definition->setAbstract(true);
+        $container->setDefinition('my-command', $definition);
+
+        $container->compile();
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The service "my-command" tagged "console.command" must be a subclass of "Symfony\Component\Console\Command\Command".
+     */
+    public function testProcessThrowAnExceptionIfTheServiceIsNotASubclassOfCommand()
+    {
+        $container = new ContainerBuilder();
+        $container->setResourceTracking(false);
+        $container->addCompilerPass(new AddConsoleCommandPass());
+
+        $definition = new Definition('SplObjectStorage');
+        $definition->addTag('console.command');
+        $container->setDefinition('my-command', $definition);
+
+        $container->compile();
+    }
+
+    public function testHttpKernelRegisterCommandsIngoreCommandAsAService()
```

Comment:
This whole test case looks a bit suspicious to me. We don't have anything HttpKernel specific here, do we?